odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/reactive/ReactiveDataSourceRepositoryImpl.java

```diff
@@ -0,0 +1,194 @@
+            conditions.add(DATA_SOURCE.IS_DELETED.isFalse());
+        }
+
+        final SelectConditionStep<Record> query = DSL
+            .select(DATA_SOURCE.asterisk())
+            .select(NAMESPACE.asterisk())
+            .select(TOKEN.asterisk())
+            .from(DATA_SOURCE)
+            .leftJoin(NAMESPACE).on(NAMESPACE.ID.eq(DATA_SOURCE.NAMESPACE_ID))
+            .leftJoin(TOKEN).on(TOKEN.ID.eq(DATA_SOURCE.TOKEN_ID))
+            .where(conditions);
+
+        return jooqReactiveOperations.flux(query).map(this::mapRecordIntoDto);
+    }
+
+    @Override
+    public Flux<DataSourceDto> listActive() {
+        final SelectConditionStep<Record> query = DSL
+            .select(DATA_SOURCE.asterisk())
+            .select(NAMESPACE.asterisk())
+            .select(TOKEN.asterisk())
+            .from(DATA_SOURCE)
+            .leftJoin(NAMESPACE).on(NAMESPACE.ID.eq(DATA_SOURCE.NAMESPACE_ID))
+            .leftJoin(TOKEN).on(TOKEN.ID.eq(DATA_SOURCE.TOKEN_ID))
+            .where(DATA_SOURCE.ACTIVE.isTrue())
+            .and(DATA_SOURCE.IS_DELETED.isFalse())
+            .and(DATA_SOURCE.CONNECTION_URL.isNotNull())
+            .and(DATA_SOURCE.CONNECTION_URL.notEqual(""));
+
+        return jooqReactiveOperations.flux(query).map(this::mapRecordIntoDto);
+    }
+
+    @Override
+    public Mono<Boolean> existsByNamespace(final long namespaceId) {
+        final Select<? extends Record1<Boolean>> query = jooqQueryHelper.selectExists(
+            DSL.selectFrom(DATA_SOURCE)
+                .where(DATA_SOURCE.NAMESPACE_ID.eq(namespaceId))
+                .and(DATA_SOURCE.IS_DELETED.isFalse())
+        );
+
+        return jooqReactiveOperations.mono(query).map(r -> r.get(0, Boolean.class));
+    }
+
+    @Override
+    public Mono<DataSourceDto> injectOddrn(final long id, final String oddrn) {
+        final UpdateResultStep<DataSourceRecord> query = DSL.update(DATA_SOURCE)
+            .set(DATA_SOURCE.ODDRN, oddrn)
+            .set(DATA_SOURCE.UPDATED_AT, LocalDateTime.now())
+            .where(DATA_SOURCE.ID.eq(id))
+            .returning();
+
+        return jooqReactiveOperations.mono(query).map(this::mapRecordIntoDto);
+    }
+
+
+    private List<Condition> queryCondition(final String nameQuery) {
+        final var conditionsList = new ArrayList<Condition>();
+
+        conditionsList.add(DATA_SOURCE.IS_DELETED.isFalse());
+        if (StringUtils.isNotEmpty(nameQuery)) {
+            conditionsList.add(DATA_SOURCE.NAME.startsWithIgnoreCase(nameQuery));
+        }
+
+        return conditionsList;
+    }
+
+    private DataSourceDto mapRecordIntoDto(final Record record, final String dataSourceCteName) {
+        final TokenPojo tokenPojo = jooqRecordHelper.extractRelation(record, TOKEN, TokenPojo.class);
+        return new DataSourceDto(
+            jooqRecordHelper.remapCte(record, dataSourceCteName, DATA_SOURCE).into(DataSourcePojo.class),
+            record.into(NAMESPACE).into(NamespacePojo.class),
```

Comment:
I believe it must be 
`jooqRecordHelper.extractRelation(record, NAMESPACE, NamespacePojo.class)`

Otherwise in case of nullable namespace it will be object with null fields